sw/device/lib/testing/test_framework/check.h

```diff
@@ -43,15 +44,36 @@
       } else {                                       \
         LOG_ERROR("CHECK-fail: " __VA_ARGS__);       \
       }                                              \
-      /* Currently, this macro will call into        \
-         the test failure code, which logs           \
-         "FAIL" and aborts. In the future,           \
-         we will try to condition on whether         \
-         or not this is a test.*/                    \
-      test_status_set(kTestStatusFailed);            \
+      _sts = INTERNAL();                             \
```

Comment:
we may eventually move the LOGs to the CHECK macro itself, so the don't print them with the use of the TRY macros, but this should be ok for now, and is a super easy change later on.